### ISSUE TITLE
Postgresql Datasource: trying to select arrays returns a string instead

### ISSUE THREAD

**Issue by Luctins** [grafana/grafana#73336]
### What happened?

When trying to implement a FFT visualizer with the heatmap plugin, SQL arrays are converted to strings in the response.

The used sql expression was:
```sql
SELECT 
  "timestamp" as "time",
  freq,
  mag
from 
  fft_ch1
WHERE
  "timestamp" >= $__timeFrom() AND "timestamp" <= $__timeTo()
```
And the table shape is:

| Column | Type |
|-- | -- |
| id |integer Auto Increment [nextval('fft_ch1_id_seq')] |  
|timestamp | timestamptz NULL [now()] |  
| mag| real[] NULL |  
| freq | real[] NULL |  

The received response was:
![image](https://github.com/grafana/grafana/assets/4743486/9e250530-f5a2-4093-8b84-063b20a26f1b)

And trying to convert it using json extraction did not work also.

I've also tried a version with unnest and a join, but it didn't work too.

Full json debug log [here](https://gist.github.com/Luctins/5545a79c73aabcd91284943d33ca931d).

### What did you expect to happen?

The array would be passed as is to grafana.

### Did this work before?

Not that I know.

### How do we reproduce it?

1. Create a array in a postgresql datasource
2. Try to select it as is.
3. It is returned as string instead of a array.

### Is the bug inside a dashboard panel?

Not really.

### Environment (with versions)?

Grafana: 9.5.2 (cfcea75916)
Browser: Firefox 116
Host OS: Ubuntu 20.04.6 LTS (Focal Fossa)
Client OS: Fedora Linux 38 (Workstation Edition)

### Grafana platform?

Docker

### Datasource(s)?

Postgre SQL

**Comment by mellieA** [comment]
@Luctins I found someone in the community that was having a similar issue and suggested flattening the array: 

https://community.grafana.com/t/plotting-a-data-array-from-postgresql-timescaledb-array/27800/2 

I'll have our dev squad that works on that data source confirm if the query response is expected to return as a string

**Comment by Luctins** [comment]
@mellieA  Thanks for the quick reply.

I have indeed seen that discussion, that's why I mentioned trying to use `unnest()`  unsuccesfully before.

At first it seeemed to work, but on close inspection did not, all my samples were merged at the bottom.

For anyone interested the posgresql code is:

```sql
SELECT 
  "timestamp" as "time",
  m, 
  fr
from 
  fft_ch1
join
  unnest(fft_ch1.mag, fft_ch1.freq) as unnested(m,fr)
on
  true
WHERE
  "timestamp" >= $__timeFrom() AND "timestamp" <= $__timeTo()
```

Where it will unroll the array into rows, but as I have understood so far, it should be unrolled into different columns to work properly. My current array is too large (around 20k elements wide) even for manually inserting `a[0]`, `a[1]`, .. , `a[n-1]` into the sql expression or using `EXECUTE`.
